### ISSUE TITLE
IsMaximizable = false but Maximize menu item was Enabled in right click Context menu on titlebar.

### ISSUE THREAD

**Issue by tera1707** [microsoft/microsoft-ui-xaml#8201]
### Describe the bug

WinUI3 packaged app.

Let the WinUI 3 window not maximizable with `IsMaximizable = false` of  Overlapped Presenter.

The Maximize button on top right of the Window turns disable, and the `Miximize` item in the context menu that appears when I right click on title bar also turns disable too.

but the `Maximize` menu item in the context menu that appeared when I right clicked on the top edge of the window title bar was enabled.

right click on the middle of the title bar

![image](https://user-images.githubusercontent.com/19180456/219694152-5c64bfe4-d464-4066-8853-309025a13191.png)

right click on the top edge of the title bar

![image](https://user-images.githubusercontent.com/19180456/219694220-00b1d923-c4b8-4d46-9b6a-4502a57addc5.png)

This is my code behind of MainWindow.

```cs
using Microsoft.UI.Windowing;
using Microsoft.UI.Xaml;

namespace MaximizeTestWinui3
{
    public sealed partial class MainWindow : Window
    {
        public MainWindow()
        {
            this.InitializeComponent();
            
            // ウインドウのハンドルを取得
            var hWnd = WinRT.Interop.WindowNative.GetWindowHandle(this);
            Microsoft.UI.WindowId windowId = Microsoft.UI.Win32Interop.GetWindowIdFromWindow(hWnd);
            Microsoft.UI.Windowing.AppWindow appWindow = Microsoft.UI.Windowing.AppWindow.GetFromWindowId(windowId);

            // 標準タイプのウインドウ(OverlappedPresenter)に対して、最大化の無効設定を行う
            var op = OverlappedPresenter.Create();
            op.IsMaximizable = false;
            op.IsResizable = false;
            appWindow.SetPresenter(op);
        }
    }
}
```

I think `IsResizable = false` may causes this problem because..
When I set `IsResizable` true, this problem did not occurred and the mouse pointer change to  arrow on the top edge of title bar. and it was not possible to show context menu with right click.

### Steps to reproduce the bug

1. Run the WinUI3 Window with my code.
2. Right click on the top edge of the window.

### Expected behavior

Show context menu like this.
![image](https://user-images.githubusercontent.com/19180456/219694152-5c64bfe4-d464-4066-8853-309025a13191.png)

### NuGet package version

WinUI 3 - Windows App SDK 1.2.3: 1.2.230118.102

### Windows version

Windows 11 (22H2): Build 22621

**Comment by DJRM2021** [comment]
Workaround:
Instead of ```var op = OverlappedPresenter.Create();```, use 
```
OverlappedPresenter windowPresenter;
windowPresenter = (OverlappedPresenter)AppWindow.Presenter;
windowPresenter.IsMaximizable = false;
windowPresenter.IsMinimizable = true;
windowPresenter.IsResizable = false;
```
![image](https://user-images.githubusercontent.com/77593544/219956198-25dee278-e575-4ebf-b20e-408bf8d3e7fc.png)

Please note that right click on the middle of the title bar and on the top edge produces the same expected result.

**Comment by tera1707** [comment]
@DJRM2021, 
Thank you for your replying.
I try your workaround, but same problem reproduced...
![Animation](https://user-images.githubusercontent.com/19180456/220112797-37c50a15-ca6d-42ee-98e4-69db8aa16df0.gif)

Like gif below.

**Comment by DJRM2021** [comment]
There's no need to set the presenter if the presenter you're trying to modify is the current one (```appWindow.Presenter```).
Try commenting the ```appWindow.SetPresenter(windowPresenter);```
![image](https://user-images.githubusercontent.com/77593544/220330143-40a74044-0bb7-4fa2-8a5b-7ae98e2720ad.png)

**Comment by tera1707** [comment]
Thank you for replying.
I commented out the line below.
"appWindow.Set Presenter(window Presenter);"

But the same still reproduced.
I guess something different between my and your environment...

**Comment by castorix** [comment]
I have the same bug on Windows 10 : if I click on the top of the Caption, the System context menu is wrong
(I tried to modifiy it with WM_INITMENUPOPUP but failed atm...)

**Comment by tera1707** [comment]
yes, I tried Workaround with `EnableMenuItem()` too, but it was failed.
I found that it was not able to grayout "default" menu item like "maximize", "minimize"...

(I succeeded to grayout items I added with `InsertMenuItem()` by myself.)

In my case, it was OK to delete "Maximize" menu item with `DeleteMenu()` func to prevent users from interacting with Maximize menu item.

**Comment by castorix** [comment]
This test worked for me (Windows 10 22H2), inside WM_INITMENUPOPUP :

                        EnableMenuItem(wParam, SC_MAXIMIZE, MF_BYCOMMAND | MF_DISABLED | MF_GRAYED);
                        EnableMenuItem(wParam, SC_SIZE, MF_BYCOMMAND | MF_DISABLED | MF_GRAYED);                     
                        return 0;

with

        [DllImport("User32.dll", SetLastError = true, CharSet = CharSet.Unicode)]
        public static extern bool EnableMenuItem(IntPtr hMenu, uint uIDEnableItem, uint uEnable);

        public const int MF_BYCOMMAND = 0x00000000;
        public const int MF_BYPOSITION = 0x00000400;
        public const int MF_ENABLED = 0x00000000;
        public const int MF_GRAYED = 0x00000001;
        public const int MF_DISABLED = 0x00000002;
        
        public const int SC_SIZE = 0xF000;
        public const int SC_MOVE = 0xF010;
        public const int SC_MINIMIZE = 0xF020;
        public const int SC_MAXIMIZE = 0xF030;
        public const int SC_NEXTWINDOW = 0xF040;
        public const int SC_PREVWINDOW = 0xF050;
        public const int SC_CLOSE = 0xF060;
        public const int SC_RESTORE = 0xF120;

**Comment by tera1707** [comment]
@castorix ,
I tried your WAY, and it worked well!
This is my WA code.

```cs
using Microsoft.UI.Windowing;
using Microsoft.UI.Xaml;
using System;
using System.Runtime.InteropServices;

namespace MaximizeTestWinui3
{
    public sealed partial class MainWindow : Window
    {
        private NativeMethods.WinProc newWndProc = null;
        private IntPtr oldWndProc = IntPtr.Zero;

        public MainWindow()
        {
            this.InitializeComponent();

            var hwnd = WinRT.Interop.WindowNative.GetWindowHandle(this);
            newWndProc = new NativeMethods.WinProc(NewWindowProc);
            oldWndProc = NativeMethods.SetWindowLongPtr64(hwnd, PInvoke.User32.WindowLongIndexFlags.GWL_WNDPROC, newWndProc);

            var windowId = Microsoft.UI.Win32Interop.GetWindowIdFromWindow(hwnd);
            var appWindow = Microsoft.UI.Windowing.AppWindow.GetFromWindowId(windowId);
            ((OverlappedPresenter)appWindow.Presenter).IsMaximizable = false;
            ((OverlappedPresenter)appWindow.Presenter).IsMinimizable = true;
            ((OverlappedPresenter)appWindow.Presenter).IsResizable = false;
        }

        private IntPtr NewWindowProc(IntPtr hWnd, PInvoke.User32.WindowMessage Msg, IntPtr wParam, IntPtr lParam)
        {
            switch (Msg)
            {
                case PInvoke.User32.WindowMessage.WM_INITMENUPOPUP:
                {
                    NativeMethods.EnableMenuItem(wParam, NativeMethods.SC_MAXIMIZE, NativeMethods.MF_BYCOMMAND | NativeMethods.MF_DISABLED | NativeMethods.MF_GRAYED); 
                    return IntPtr.Zero;
                }
            }
            return NativeMethods.CallWindowProc(oldWndProc, hWnd, Msg, wParam, lParam);
        }
    }

    public class NativeMethods
    {
        public delegate IntPtr WinProc(IntPtr hWnd, PInvoke.User32.WindowMessage Msg, IntPtr wParam, IntPtr lParam);

        [DllImport("user32.dll", EntryPoint = "SetWindowLongPtr")]
        public static extern IntPtr SetWindowLongPtr64(IntPtr hWnd, PInvoke.User32.WindowLongIndexFlags nIndex, WinProc dwNewLong);
        
        [DllImport("user32.dll")]
        public static extern IntPtr CallWindowProc(IntPtr lpPrevWndFunc, IntPtr hWnd, PInvoke.User32.WindowMessage Msg, IntPtr wParam, IntPtr lParam);

        [DllImport("User32.dll", SetLastError = true, CharSet = CharSet.Unicode)]
        public static extern bool EnableMenuItem(IntPtr hMenu, uint uIDEnableItem, uint uEnable);

        public const int MF_BYCOMMAND = 0x00000000;
        public const int MF_ENABLED = 0x00000000;
        public const int MF_GRAYED = 0x00000001;
        public const int MF_DISABLED = 0x00000002;

        public const int SC_MAXIMIZE = 0xF030;
    }
}
```
